src/resolve-system-dependencies/resolve-system-dependencies.cc

```diff
@@ -0,0 +1,202 @@
+#include "derivations.hh"
+#include "globals.hh"
+#include "shared.hh"
+#include "store-api.hh"
+#include <sys/utsname.h>
+#include <algorithm>
+#include <iostream>
+#include <fstream>
+#include <mach-o/loader.h>
+#include <mach-o/swap.h>
+
+using namespace nix;
+
+typedef std::map<string, std::set<string>> SetMap;
+
+static auto cacheDir = Path{};
+
+Path resolveCacheFile(const Path & lib) {
+    Path lib2 = Path(lib);
```

Comment:
Instead of using a const reference parameter and doing explicit copying, you could directly accept the lib argument via value.